src/v2/Apps/Conversation/Components/Conversation.tsx

```diff
@@ -175,6 +61,55 @@ const Conversation: React.FC<ConversationProps> = props => {
     />
   ))
 
+  // Pagination Scroll Logic
+  const [fetchingMore, setFetchingMore] = useState(false)
+  const [messagesTop, setMessagesTop] = useState(null)
+  const scrollContainer = useRef(null)
+
+  useEffect(() => {
+    if (!("IntersectionObserver" in window)) return
+    const observer = new IntersectionObserver(
+      entries => {
+        const first = entries[0]
+        if (first.isIntersecting && !initialMount.current) {
+          _loadMore()
+        }
+      },
+      { threshold: 0, rootMargin: `150px` }
+    )
+
+    const currentMessagesTop = messagesTop
+    const currentObserver = observer
```

Comment:
**question:** Are these variables being rebound just for added clarity?

<!--
Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.
-->

Reply:
Hmm, I set the first one up initially to create a closure and have it available to the cleanup effect function, and the second one for clarity (also initially). But I've refactored since then, and these might not be needed any longer. Checking.